### PR TITLE
Fix routerinformationupdate can take complex json

### PR DESCRIPTION
**Pull request by chunhtai**
Fixes https://github.com/flutter/flutter/issues/87823

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: lib/web_ui/lib/src/engine/window.dart [hunk 1]
```diff
@@ -171,7 +171,7 @@ class EngineFlutterWindow extends ui.SingletonFlutterWindow {
           assert(arguments != null);
           browserHistory.setRouteName(
             arguments!.tryString('location'),
-            state: arguments.tryString('state'),
+            state: arguments['state'],
```
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/window.dart:174]: @yjbanov should we add a `tryObject()`?

**Review by mdebbar — APPROVED** [review]
LGTM

File: lib/web_ui/lib/src/engine/window.dart (hunk 1 above)
Comment (chunhtai) [reply lib/web_ui/lib/src/engine/window.dart:174]: what is the point of tryObject though? it is already an object?

Comment (mdebbar) [reply lib/web_ui/lib/src/engine/window.dart:174]: Without a type, it's `dynamic` not `Object?`. Not sure if that's an important distinction or not.

Comment (mdebbar) [reply lib/web_ui/lib/src/engine/window.dart:174]: Btw, feel free to submit the PR. If we need to change it to `Object?` we can do so later.

Comment (chunhtai) [reply lib/web_ui/lib/src/engine/window.dart:174]: I can add that if that makes sense, btw what is the motive to avoid `as String?` and use `tryString`

**Pull request merged**